pkgs/tools/virtualization/dockstarter/default.nix

```diff
@@ -0,0 +1,43 @@
+{ bash
+, coreutils
+, fetchFromGitHub
+, git
+, lib
+, makeWrapper
+, ncurses
+, stdenv
+, sudo
+}:
+stdenv.mkDerivation rec {
+  pname = "dockstarter";
+  version = "unstable-2022-10-26";
+
+  src = fetchFromGitHub {
+    owner = "ghostwriters";
+    repo = pname;
+    rev = "a1b6b6e29aa135c2a61ea67ca05e9e034856ca08";
+    hash = "sha256-G26DFme6YaizdE5oHBo/IqV+1quu07Bp+IykXtO/GgA=";
+  };
+
+  doBuild = false;
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  installPhase = ''
+    install -Dm755 main.sh $out/bin/ds
+    wrapProgram $out/bin/ds --prefix PATH : ${lib.makeBinPath ((lib.optionals (!stdenv.isDarwin) [ sudo ]) ++ [
```

Comment:
Adding sudo like this does not work. It must use the wrapper.

Reply:
dockstarter is a shell script that calls sudo inside one of its methods, it is not intended to be run as root or via `sudo dockstarter`; do you have an example of the wrapper you speak of? (I am unsure what change is being asked for.)

Reply:
On nixos `which sudo` should usually point to `/run/wrappers/bin/sudo` which is created here:  https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/security/wrappers/wrapper.nix

The sudo package doesn't work out of the box this way.

Reply:
as this package is targeting multiple systems: nixos, darwin, etc. what is the best way to ensure that `sudo` exists on the system? or is it simply required to be an impure dependency that gets pulled in via `PATH`? (I know this is a requirement on darwin.)

Reply:
> what is the best way to ensure that `sudo` exists on the system?

There is none because it is not possible. Binaries in the nix store are not allowed to have the suid bit that is required for sudo to work.



> or is it simply required to be an impure dependency that gets pulled in via `PATH`?

Yes

---

Please remove sudo from the wrapper.

Reply:
happily; should I track the sudo dependency at all?

Reply:
> happily; should I track the sudo dependency at all?

I'm not 100% sure but I do not think you should.